modules/animationqt/src/sequenceeditor/propertysequenceeditor.cpp

```diff
@@ -193,17 +190,16 @@ PropertySequenceEditor::PropertySequenceEditor(KeyframeSequence& sequence, Track
     sublayout->addWidget(new QLabel("Interpolation"), 0, 0);
     sublayout->addWidget(interpolation_, 0, 1);
 
-    auto map = manager.getInterpolationMapping();
-
-    for (auto range = map.equal_range(bpt.getProperty()->getClassIdentifier());
-         range.first != range.second; ++range.first) {
-        auto ip = manager.getInterpolationFactory().create(range.first->second);
+    auto& valseq = dynamic_cast<ValueKeyframeSequence&>(sequence);
+    for (auto interpObj : valseq.getSupportedInterpolations(manager.getInterpolationFactory())) {
+        auto ip = interpObj->create();
```

Comment:
at some point the name should be in the factory object, so we don't have to create them just to find the name...

Reply:
Agree, but not a priority